dataloadgen.go

```diff
@@ -244,14 +254,21 @@ func (l *Loader[KeyT, ValueT]) addKeyToBatch(b *loaderBatch[KeyT, ValueT], key K
 
 	if l.maxBatch != 0 && pos >= l.maxBatch-1 {
 		ctxs := l.batch.contexts
+		spans := l.batch.spans
 		b.fetchExecuted = true
 		l.batch = nil
 		go func(l *Loader[KeyT, ValueT], ctxs []context.Context) {
 			for _, ctx := range ctxs {
 				_, span := l.tracer.Start(ctx, "dataloadgen.fetch.keylimit")
 				defer span.End()
 			}
+
 			b.results, b.errors = l.fetch(b.keys)
+
+			for _, span := range spans {
```

Comment:
Same issue here

Reply:
Is accessing these lists in the goroutines safe? Can't elements be added to them concurrently? Let's make sure to run the tests with -race

Reply:
it looks like these blocks are protected by the mutex, no? 
tests with `-race` pass, however there's no tracer so that doesn't tell us much

Reply:
I see why this works. The only other thing that reads/writes the batch struct runs only after `close(b.done)` runs. That's why there's no race.